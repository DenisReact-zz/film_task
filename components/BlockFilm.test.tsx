import { fireEvent, render, screen } from "@testing-library/react";
import BlockFilm from "./BlockFilm";

describe("tests for block film", () => {
  const mockFilm = {
    adult: false,
    backdrop_path: "/iNh3BivHyg5sQRPP1KOkzguEX0H.jpg",
    genre_ids: [18, 80],
    id: 278,
    original_language: "en",
    original_title: "The Shawshank Redemption",
    overview:
      "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
    popularity: 64.924,
    poster_path: "/q6y0Go1tsGEsmtFryDOJo3dEmqu.jpg",
    release_date: "1994-09-23",
    title: "The Shawshank Redemption",
    video: false,
    vote_average: 8.7,
    vote_count: 21437,
  };
  beforeEach(() => {
    render(<BlockFilm dataFilm={mockFilm} />);
  });
  it("check layout block film", () => {
    const blockFilm = screen.getByTestId("blockFilm");
    expect(blockFilm).toMatchSnapshot();
  });
  it("correct link", () => {
    const link: HTMLLinkElement = screen.getByTestId("linkBlockFilm");
    expect(/themoviedb.org/.test(link.href)).toEqual(true);
  });
  it("event on star", () => {
    const star = screen.getByTestId("star");
    expect(star.onclick).not.toBeUndefined();
  });
  it("add value in localStorage on click of star", () => {
    const star = screen.getByTestId("star");
    fireEvent.click(star);
    expect(window.localStorage.getItem("markedFilms")).not.toBeNull();
  });
});
