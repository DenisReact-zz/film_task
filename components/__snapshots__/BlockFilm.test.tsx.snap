// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for block film check layout block film 1`] = `
<div
  class="blockFilm"
  data-testid="blockFilm"
>
  <div
    class="img"
  >
    <div
      class="starLink"
    >
      <a
        data-testid="linkBlockFilm"
        href="https://www.themoviedb.org/movie/278"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </a>
      <div
        class="star"
        data-testid="star"
      >
        <svg
          enable-background="new 0 0 32 32"
          id="Glyph"
          version="1.1"
          viewBox="0 0 32 32"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M29.895,12.52c-0.235-0.704-0.829-1.209-1.549-1.319l-7.309-1.095l-3.29-6.984C17.42,2.43,16.751,2,16,2 s-1.42,0.43-1.747,1.122l-3.242,6.959l-7.357,1.12c-0.72,0.11-1.313,0.615-1.549,1.319c-0.241,0.723-0.063,1.507,0.465,2.046 l5.321,5.446l-1.257,7.676c-0.125,0.767,0.185,1.518,0.811,1.959c0.602,0.427,1.376,0.469,2.02,0.114l6.489-3.624l6.581,3.624 c0.646,0.355,1.418,0.311,2.02-0.114c0.626-0.441,0.937-1.192,0.811-1.959l-1.259-7.686l5.323-5.436 C29.958,14.027,30.136,13.243,29.895,12.52z"
            id="XMLID_328_"
          />
        </svg>
      </div>
    </div>
    <div
      class="rating"
    >
      <p>
        8.7
      </p>
    </div>
  </div>
  <div
    class="text"
  >
    <h4>
      The Shawshank Redemption
    </h4>
    <p>
      1994-09-23
    </p>
  </div>
</div>
`;
